ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStore.java

```diff
@@ -480,12 +527,21 @@ private boolean checkCacheState(String blobKey, BlobState... desiredStates) {
   /**
    * Add a blob state mapping to the recent blob cache.
    * @param blobKey the blob key to cache.
+   * @param lifeVersion life version to cache.
    * @param blobState the state of the blob.
    */
-  // Visible for test
-  void addToCache(String blobKey, BlobState blobState) {
+  // Visible for test.
+  void addToCache(String blobKey, short lifeVersion, BlobState blobState) {
     if (isVcr) {
-      recentBlobCache.put(blobKey, blobState);
+      if (blobState == BlobState.TTL_UPDATED) {
+        // In case of ttl update we update the ttl without taking into account the life version.
+        // So make sure that we do not decrease the lifeVersion in cache due to an incoming ttl update.
+        if (recentBlobCache.containsKey(blobKey)) {
+          lifeVersion = (lifeVersion > recentBlobCache.get(blobKey).getLifeVersion()) ? lifeVersion
+              : recentBlobCache.get(blobKey).getLifeVersion();
```

Comment:
Can this just be Math.max(lifeVersion, recentBlobCache.get(blobKey).getLifeVersion()) ?